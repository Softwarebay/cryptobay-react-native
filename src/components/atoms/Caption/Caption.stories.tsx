import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { Caption } from './Caption';

storiesOf('Atoms/Caption', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Gray Caption', () => <Caption value={text('value', '$9.58799')} />)
  .add('White Caption', () => (
    <Caption value={text('value', 'Mar 3 at 10:04am')} variant="white" />
  ));
