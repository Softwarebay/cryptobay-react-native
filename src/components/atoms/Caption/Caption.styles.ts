import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  caption: {
    fontFamily: 'Poppins',
    fontSize: 12,
    lineHeight: 18,
    fontWeight: '400',
  },
  gray: {
    color: '#ABAFC4',
  },
  white: {
    color: '#FFFFFF',
  },
});
