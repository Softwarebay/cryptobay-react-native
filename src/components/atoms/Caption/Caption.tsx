import React from 'react';
import { Text, TextStyle } from 'react-native';

import { styles } from './Caption.styles';

type CaptionProps = {
  value: string;
  variant?: 'gray' | 'white';
  style?: TextStyle;
  testID?: string;
};

export const Caption: React.FC<CaptionProps> = ({
  value,
  variant = 'gray',
  style,
  testID,
}: CaptionProps) => {
  return (
    <Text testID={testID} style={[styles.caption, styles[variant], style]}>
      {value}
    </Text>
  );
};
