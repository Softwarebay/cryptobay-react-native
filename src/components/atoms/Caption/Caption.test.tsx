import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Caption } from './Caption';

describe('Caption Atom', () => {
  test('Coponent Snapshot', () => {
    const tree = renderer.create(<Caption value="$9.58799" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure variant add specific color style rule', () => {
    const { rerender } = render(
      <Caption value="$9.58799" testID="atom-caption" />,
    );
    const caption = screen.getByTestId('atom-caption');
    expect(caption).toHaveTextContent('$9.58799');
    expect(caption).toHaveStyle({ color: '#ABAFC4' });
    rerender(
      <Caption
        value="Mar 3 at 10:04am"
        variant="white"
        testID="atom-caption"
      />,
    );
    expect(caption).toHaveTextContent('Mar 3 at 10:04am');
    expect(caption).toHaveStyle({ color: '#FFFFFF' });
  });
});
