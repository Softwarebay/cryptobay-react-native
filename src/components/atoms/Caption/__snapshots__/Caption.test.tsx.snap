// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caption Atom Coponent Snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "fontFamily": "Poppins",
        "fontSize": 12,
        "fontWeight": "400",
        "lineHeight": 18,
      },
      Object {
        "color": "#ABAFC4",
      },
      undefined,
    ]
  }
>
  $9.58799
</Text>
`;
